kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/FilterBuilderImpl.java

```diff
@@ -0,0 +1,203 @@
+  public FilterBuilder<FilterWatchListDeletable<T, L>> withoutField(String key, String value) {
+    context.fieldsNot.merge(key, new String[]{value}, (oldList, newList) -> {
+      if (Utils.isNotNullOrEmpty(newList[0])) { // Only add new values when not null
+        final String[] concatList = (String[]) Array.newInstance(String.class, oldList.length + newList.length);
+        System.arraycopy(oldList, 0, concatList, 0, oldList.length);
+        System.arraycopy(newList, 0, concatList, oldList.length, newList.length);
+        return concatList;
+      } else {
+        return oldList;
+      }
+    });
+    return this;
+  }
+
+  @Override
+  public FilterBuilder<FilterWatchListDeletable<T, L>> withLabelSelector(LabelSelector selector) {
+    Map<String, String> matchLabels = selector.getMatchLabels();
+    if (matchLabels != null) {
+      withLabels(matchLabels);
+    }
+    List<LabelSelectorRequirement> matchExpressions = selector.getMatchExpressions();
+    if (matchExpressions != null) {
+      for (LabelSelectorRequirement req : matchExpressions) {
+        String operator = req.getOperator();
+        String key = req.getKey();
+        switch (operator) {
+          case "In":
+            withLabelIn(key, req.getValues().toArray(new String[]{}));
+            break;
+          case "NotIn":
+            withLabelNotIn(key, req.getValues().toArray(new String[]{}));
+            break;
+          case "DoesNotExist":
+            withoutLabel(key);
+            break;
+          case "Exists":
+            withLabel(key);
+            break;
+          default:
+            throw new IllegalArgumentException("Unsupported operator: " + operator);
+        }
+      }
+    }
+    return this;
+  }
+
+  @Override
+  public FilterBuilder<FilterWatchListDeletable<T, L>> withInvolvedObject(ObjectReference objectReference) {
+    if (objectReference.getName() != null) {
+      context.fields.put(INVOLVED_OBJECT_NAME, objectReference.getName());
+    }
+    if (objectReference.getNamespace() != null) {
+      context.fields.put(INVOLVED_OBJECT_NAMESPACE, objectReference.getNamespace());
+    }
+    if (objectReference.getKind() != null) {
+      context.fields.put(INVOLVED_OBJECT_KIND, objectReference.getKind());
+    }
+    if (objectReference.getUid() != null) {
+      context.fields.put(INVOLVED_OBJECT_UID, objectReference.getUid());
+    }
+    if (objectReference.getResourceVersion() != null) {
+      context.fields.put(INVOLVED_OBJECT_RESOURCE_VERSION, objectReference.getResourceVersion());
+    }
+    if (objectReference.getApiVersion() != null) {
+      context.fields.put(INVOLVED_OBJECT_API_VERSION, objectReference.getApiVersion());
+    }
+    if (objectReference.getFieldPath() != null) {
+      context.fields.put(INVOLVED_OBJECT_FIELD_PATH, objectReference.getFieldPath());
+    }
+    return this;
+  }
+
+  @Override
+  public FilterWatchListDeletable<T, L> build() {
+    return this.baseOperation.newInstance(context);
```

Comment:
Umm, In my opinion this can be a bit confusing:
```java
client.apps().deployments().withFilter().withLabel("key1", "value1").build().list();
```
I think maybe if we name this to something like this it might be better:
```java
client.apps().deployments().withNewFilter().withLabel("key1", "value1").endFilter().list();
```

Reply:
Yes, this seems also more consistent with the rest of builders generated by sundr.io

Reply:
Right, I should have gone with nested rather than a builder.  I've committed a change to nested.